components/brave_wallet/browser/zcash/zcash_wallet_service_tasks.cc

```diff
@@ -339,12 +346,250 @@ bool CreateTransparentTransactionTask::PrepareOutputs() {
   }
 
   CHECK(OutputAddressSupported(change_address->address_string, IsTestnet()));
-  auto& change_output = transaction_.outputs().emplace_back();
+  auto& change_output = transaction_.transparent_part().outputs.emplace_back();
   change_output.address = change_address_->address_string;
   change_output.amount = change_amount;
   change_output.script_pubkey =
       ZCashAddressToScriptPubkey(change_output.address, IsTestnet());
   return true;
 }
 
+#if BUILDFLAG(ENABLE_ORCHARD)
+
+CreateShieldAllTransactionTask::CreateShieldAllTransactionTask(
+    ZCashWalletService* zcash_wallet_service,
+    const std::string& chain_id,
+    const mojom::AccountIdPtr& account_id,
+    ZCashWalletService::CreateTransactionCallback callback,
+    std::optional<uint64_t> random_seed_for_testing)
+    : zcash_wallet_service_(zcash_wallet_service),
+      chain_id_(chain_id),
+      account_id_(account_id.Clone()),
+      callback_(std::move(callback)),
+      random_seed_for_testing_(random_seed_for_testing) {}
+
+CreateShieldAllTransactionTask::~CreateShieldAllTransactionTask() = default;
+
+::rust::Box<orchard::OrchardUnauthorizedBundleResult>
+CreateShieldAllTransactionTask::CreateOrchardUnauthorizedBundle(
+    ::rust::Slice<const uint8_t> tree_state,
+    ::rust::Vec<orchard::OrchardOutput> outputs) {
+  if (random_seed_for_testing_) {
+    CHECK_IS_TEST();
+    return create_testing_orchard_bundle(std::move(tree_state),
+                                         std::move(outputs),
+                                         random_seed_for_testing_.value());
+  } else {
+    return create_orchard_bundle(std::move(tree_state), std::move(outputs));
+  }
+}
+
+void CreateShieldAllTransactionTask::ScheduleWorkOnTask() {
+  if (error_) {
+    std::move(callback_).Run(base::unexpected(error_.value()));
+    return;
+  }
+
+  if (!tree_state_) {
+    GetTreeState();
+    return;
+  }
+
+  if (!utxo_map_) {
+    GetAllUtxos();
+    return;
+  }
+
+  if (!chain_height_) {
+    GetChainHeight();
+    return;
+  }
+
+  if (!CreateTransaction()) {
+    std::move(callback_).Run(base::unexpected(error_.value()));
+    return;
+  }
+
+  // TODO(cypt4): This call should be async
```

Comment:
which call should be async? Why is not async now and is there a follow-up issue to fix this?